### ISSUE TITLE
Question: how to integrate with spell checking?

### ISSUE THREAD

**Issue by infokiller** [dense-analysis/ale#2773]
Thanks for this great plugin!

I often like to spell check source files I'm working on. I do this using vim's built-in spell checker and https://github.com/inkarkat/vim-SpellCheck which populates the location list with spelling errors.

This causes a race with ALE when it's available, which uses the location list for its linter errors. For example, if I enable spell checking with vim-SpellCheck and then I save the file, ALE runs and then overrides the location list entries.

Questions:
- How can I best use spell checking with ALE?
- I tried to use text linters in code files but couldn't make it work. For example, I set:

  ```vim
  let g:ale_linters['vim'] = ['alex', 'languagetool', 'proselint', 'redpen', 'remark_lint', 'textlint', 'vale', 'writegood']
  ```

  However, it didn't seem to work (I checked `:ALEInfo`). Is this a limitation, or am I doing something wrong?

Thanks!

**Comment by w0rp** [comment]
If you want to use text linters for another filetype like `vim`, try using `g:ale_linter_aliases` with `'vim': ['vim', 'text']`, so ALE will run both Vim and text linters on Vim files. Some other filetypes already use some of the text linters where they are known to work.

**Comment by w0rp** [comment]
For getting ALE and another tool which populates the locaiton list to work together, I recommend modifying the tool to support ALE's API for giving a loclist to ALE, which will then display the errors in the pretty ways ALE displays them. See `:help ale-lint-other-sources`.

**Comment by infokiller** [comment]
So I went ahead and tried implementing support today in [my fork  of vim-SpellCheck](https://github.com/infokiller/vim-SpellCheck/tree/add-ale-support).

The errors show up when I call `ale#other_source#ShowResults`, but the results are not refreshed. I tried setting up an autocmd for `ALEWantResults` as following:

```vim
augroup SpellCheckALE
	autocmd!
	autocmd User ALEWantResults call SpellCheck#quickfix#CheckForALE()
 augroup END
```

However, it seems that I can't make this event trigger. ALE's documentation notes that `This User autocommand is triggered before ALE begins a lint cycle`.
Is there a way for me to manually trigger the lint cycle? I tried editing the file and saving it, as well as using only the default ALE options, but it seems `SpellCheck#quickfix#CheckForALE()` is not being called.

Thanks!
